lib/signaling/v2/peerconnection.js

```diff
@@ -290,7 +299,8 @@ class PeerConnectionV2 extends StateMachine {
    * @property {RTCIceConnectionState}
    */
   get iceConnectionState() {
-    return this._peerConnection.iceConnectionState;
+    return (this._isIceConnectionInactive && this._peerConnection.iceConnectionState === 'disconnected')
```

Comment:
@makarandp0 Nitpick: I don't think we need the `()` around the condition. Also, can you indent the `?` with just two spaces after `return` like this:
```js
return this._isIceConnectionInactive && this._peerConnection.iceConnectionState === 'disconnected'
  ? 'failed' : this._peerConnection.iceConnectionState;
```
?